### PR TITLE
Adding quasielasticbayes

### PR DESCRIPTION
**Pull request by robertapplin**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by robertapplin** [comment]
I confirm I am happy to be listed as a maintainer

**Comment by robertapplin** [comment]
@conda-forge/help-python, ready for review!

File: recipes/quasielasticbayes/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+build:
+  skip: True  # [py<310 or py>310 or osx]
+  script: {{ PYTHON }} setup.py install
+  number: 0
+
+requirements:
+  build:
+    - m2w64-toolchain_win-64  # [win]
+    - gfortran                # [linux]
```
[the remark this answers is not in the document]
Comment (robertapplin) [reply recipes/quasielasticbayes/meta.yaml:20]: Thanks for the review! I seem to get different results in my unit tests when I use `- {{ compiler('fortran') }}` for linux, and I get the following error for both `- {{ compiler('fortran') }}` and `- {{ compiler('m2w64_fortran') }}` on Windows
```
Encountered problems while solving:
  - nothing provides requested m2w64_fortran_win-64
```

[the remark this answers is not in the document]
Comment (robertapplin) [reply recipes/quasielasticbayes/meta.yaml]: Sure, latest commit tries with `{{ compiler('fortran') }}`

[the remark this answers is not in the document]
Comment (ocefpaf) [reply recipes/quasielasticbayes/meta.yaml]: Looks like numerical errors in the tests like `Mismatched elements: 30 / 16384 (0.183%)`. The fact that those are gone when you don't use conda-forge compilers using the jinja syntax is probably b/c you are getting a different version of gfortran. We should investigate that... Also, I'd report upstream why the version in this commit causes the mismatch.

[the remark this answers is not in the document]
Comment (robertapplin) [reply recipes/quasielasticbayes/meta.yaml]: What does `{{ compiler('fortran') }}` evaluate to on Windows? On my local machine it seems to evaluate to `gfortran_win-64`, but a package by this name doesn't seem to be available on conda-forge. So I'm confused why it works on the Azure windows pipeline below - maybe I'm missing some piece of information about this?

[the remark this answers is not in the document]
Comment (robertapplin) [reply recipes/quasielasticbayes/meta.yaml]: The package relies on old fortran code which requires a very specific compiler version to reproduce the expected fitting results.

I've tried to find the build environment package differences between the expected results, and the numerical differences on Linux. The only differences I see:

To achieve the correct results:
```
gcc                       12.3.0               h95e488c_3    conda-forge
gfortran                  12.3.0               h7389182_3    conda-forge
```

When we see the numerical differences:
```
binutils_linux-64         2.40                 hdade7a5_3    conda-forge
gcc_linux-64              12.3.0               h6477408_3    conda-forge
gfortran_linux-64         12.3.0               h617cb40_3    conda-forge
```

The difference between `gcc` / `gcc_linux-64` and `gfortran` / `gfortran_linux-64` is not clear to me. The numerical differences are too large to ignore on Windows it seems.

Would it alleviate your concern if I were to move the compiler packages to the `conda_build_config.yaml` file so they are not hard-coded in the `meta.yaml`?

[the remark this answers is not in the document]
Comment (ocefpaf) [reply recipes/quasielasticbayes/meta.yaml]: Sure. But let me ping @isuruf here, our Fortran wizard, for some guidance. He may have a simple solution for this.

File: recipes/quasielasticbayes/conda_build_config.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+fortran_compiler:
+  - m2w64-toolchain_win-64                 # [win]
+  - gfortran                               # [linux]
```
Comment (ocefpaf) [inline recipes/quasielasticbayes/conda_build_config.yaml:3]: Based on what you wrote here you need to pin these, no?

Comment (robertapplin) [reply recipes/quasielasticbayes/conda_build_config.yaml:3]: Thanks, I think that would be a good idea

**Pull request merged**